### PR TITLE
fix(semver): allow usage of different preids with prerelease

### PR DESCRIPTION
**Pull request by cakeinpanic**

### CODE REVIEW

File: packages/semver/src/executors/version/index.e2e.spec.ts [hunk 1]
```diff
@@ -1085,6 +1086,40 @@ $`)
...
+          version: 'prerelease',
+          preid: 'alpha',
+        },
+       fakeContext
+      );
+
+      expect(
+        (await lastValueFrom(readPackageJson('packages/a'))).version
+      ).toEqual('0.0.1-alpha.1');
+    });
```
Comment (edbzn) [inline packages/semver/src/executors/version/index.e2e.spec.ts:1120]: I would suggest using custom --preid for these new tests, like the one you mentioned in the issue (`add-feature`, `fix-bug`) and also moving them out in another describe function.

File: packages/semver/src/executors/version/utils/get-last-version.spec.ts [hunk 2]
```diff
@@ -31,16 +31,20 @@ describe(getLastVersion.name, () => {
     expect(tag).toEqual('2.1.0');
   });
 
-  it('should compute current version from previous semver prerelease tag', async () => {
+  it('should compute current version from previous semver prerelease tag with corresponding preid', async () => {
     mockGitSemverTags.mockResolvedValue([
-      'my-lib-2.1.0-beta.0',
+      'my-lib-2.1.0-beta.5',
+      'my-lib-2.1.0-alpha.0',
```
Comment (edbzn) [inline packages/semver/src/executors/version/utils/get-last-version.spec.ts:37]: Is it possible to unit test custom pre ids as well like `add-feature` and `fix-bug`.

**Review by edbzn — COMMENTED** [review]
Hi, this looks good, just left a few comments regarding the tests, I think we can improve them a bit. The [precommit](https://github.com/jscutlery/semver/blob/main/.husky/pre-commit) hook that formats project files seems to fail for you. Did you see something special logged when committing?

**Comment by cakeinpanic** [comment]
@edbzn weird but husky did not inititalize for me. I guess it'be becasue you have it in `prepack` script and I never called `npm publish`. Worth moving it to `prepare` script

File: packages/semver/src/executors/version/utils/get-last-version.spec.ts (hunk 2 above)
Comment (cakeinpanic) [reply packages/semver/src/executors/version/utils/get-last-version.spec.ts:37]: sure, no prob

**Review by edbzn — APPROVED** [review]

**Pull request merged**